src/components/frame/v5/pages/DashboardPage/partials/Menu/types.ts

```diff
@@ -0,0 +1,37 @@
+import { MessageDescriptor } from 'react-intl';
+
+export const MENU_ITEM_TYPE = {
+  BUTTON: 'button',
+  BUTTON_WITH_TOOLTIP: 'button-with-tooltip',
+  LINK: 'link',
+  DIVIDER: 'divider',
+} as const;
+
+export type MenuItemType = (typeof MENU_ITEM_TYPE)[keyof typeof MENU_ITEM_TYPE];
+
+export interface MenuItemList {
+  itemKey: string;
+  componentType: MenuItemType;
+  url?: string;
+  iconName?: string;
+  title?: MessageDescriptor | string;
+  ariaExpanded?: boolean;
+  ariaControls?: string;
+  additionalContent?: JSX.Element;
+  ariaLabel?: MessageDescriptor | string;
+  isSuccess?: boolean;
+  onClick?: () => void;
+  tooltipContent?: JSX.Element;
+  className?: string;
+}
+
+export interface MenuItemListProps {
+  key?: string;
+  items: MenuItemList[];
+}
+
+export interface UseMenuReturnType {
+  menuItems: MenuItemListProps[];
+  activeSubmenu: boolean;
```

Comment:
activeSubmenu sounds like you wanted to pass an array or a component and not a boolean